src/json.ts

```diff
@@ -177,10 +177,10 @@ export type JsonRpcError = OptionalField<
   'data'
 >;
 
-export const JsonRpcParamsStruct = optional(
-  union([record(string(), JsonStruct), array(JsonStruct)]),
-);
-export type JsonRpcParams = Infer<typeof JsonRpcParamsStruct>;
+export const JsonRpcParamsStruct: Struct<Json[] | Record<string, Json>, null> =
+  optional(union([record(string(), JsonStruct), array(JsonStruct)])) as any;
```

Comment:
Why do we require a cast here? Also wont adding `optional` allow `undefined` values? 🤔

Reply:
cc @Mrtenz

Reply:
I understand but if we use `optional` our validation will not fail if params are directly to `undefined` which is invalid? https://docs.superstructjs.org/api-reference/types#optional

Reply:
I misunderstood this change, we already allow `optional`. Which we shouldn't but that isn't what this PR is fixing.